ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,952 @@
+                Ok(())
+            }
+        }
+    }
+
+    /// Sets the region in which auto exposure should be based on.
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - Starting x coordinate of the window
+    /// * `y` - Starting y coordinate of the window
+    /// * `width` - Width of the window
+    /// * `height` - Height of the window
+    fn set_auto_exposure_window(
+        &mut self,
+        x: i16,
+        y: i16,
+        width: i16,
+        height: i16,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetAutoExposureWindow(self.camera_as_raw(), x, y, width, height);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    /// Sets the region in which auto white balance should be based on.
+    ///
+    /// # Arguments
+    ///
+    /// * `x` - Starting x coordinate of the window
+    /// * `y` - Starting y coordinate of the window
+    /// * `width` - Width of the window
+    /// * `height` - Height of the window
+    fn set_auto_white_balance_window(
+        &mut self,
+        x: i16,
+        y: i16,
+        width: i16,
+        height: i16,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r =
+                ctru_sys::CAMU_SetAutoWhiteBalanceWindow(self.camera_as_raw(), x, y, width, height);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    /// Sets whether the noise filter should be enabled or disabled for the camera
+    fn set_noise_filter(&mut self, enabled: bool) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetNoiseFilter(self.camera_as_raw(), enabled);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    /// Sets the image quality calibration data for the camera based on the passed in
+    /// [ImageQualityCalibrationData] argument
+    fn set_image_quality_calibration_data(
+        &mut self,
+        data: ImageQualityCalibrationData,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetImageQualityCalibrationData(data.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    /// Returns the current [ImageQualityCalibrationData] for the camera
+    fn get_image_quality_calibration_data(&self) -> crate::Result<ImageQualityCalibrationData> {
+        unsafe {
+            let mut data = ImageQualityCalibrationData::default();
+            let r = ctru_sys::CAMU_GetImageQualityCalibrationData(&mut data.0);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(data)
+            }
+        }
+    }
+
+    /// Sets the camera as the current sleep camera
+    // TODO: Explain sleep camera
```

Comment:
There's a to-do comment here

Reply:
I put a todo because I personally was unable to figure out what the sleep camera really does. If there is knowledge on this I can add the information.

Reply:
Oh no problem, I'll resolve this then.